pkgs/development/java-modules/maven-minimal.nix

```diff
@@ -0,0 +1,147 @@
+{ stdenv, pkgs, maven }:
+
+with stdenv.lib;
+with pkgs.javaPackages;
+
+let
+  collections = import ./collections.nix { inherit pkgs; };
+  fetchMaven = pkgs.callPackage ./m2install.nix { };
+  plugins = import ./mavenPlugins.nix { inherit stdenv pkgs maven; };
+  poms = import ./poms.nix { inherit fetchMaven; };
+in rec {
+  # Maven needs all of these to function
+  mavenMinimal = flatten
+    collections.mavenLibs_2_0_6
+    ++ collections.mavenLibs_2_0_9
+    ++ collections.mavenLibs_2_2_1
+    ++ [
+      classworlds_1_1_alpha2
+      classworlds_1_1
+      commonsCli_1_0
+      commonsLang_2_1
+      commonsLang3_3_1
+      commonsLoggingApi_1_1
+      findbugsJsr305_2_0_1
+      googleCollections_1_0
+      junit_3_8_1
```

Comment:
IMO `maven` shouldn't use `junit` at runtime at all. On a first glance it could be `test dependency` of https://mvnrepository.com/artifact/org.apache.maven.surefire/maven-surefire-common/2.17
Anyway, it's ok if it doesn't break it.
We could try to optimize it later. It may be worth to dig into `mvn dependency:tree` and treat `compile`, `test`, `provided` scopes differently instead of listing `~/.m2` tree.